client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
+	getDisplayedTimeForLabel( noPrefix = false ) {
 		const moment = this.props.moment;
-
 		const now = moment();
+		const scheduledDate = moment( this.getTimestamp() );
 
-		if ( ! post ) {
-			// Placeholder text: "a few seconds ago" in English locale
-			return now.fromNow();
-		}
+		const isScheduledPost = this.props.post.status === 'future';
+
+		let scheduledTime;
+		if ( isScheduledPost ) {
+			scheduledTime = scheduledDate.calendar( null, {
+				nextDay: this.props.translate( '[for tomorrow at] LT', {
+					comment: 'LT refers to time (eg. 18:00)',
+				} ),
+				sameElse: this.props.translate( '[for] ll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" and "for" is translated',
+				} ),
+			} );
+		} else {
+			if ( Math.abs( now.diff( this.getTimestamp(), 'days' ) ) < 7 ) {
+				const time = moment( this.getTimestamp() );
+				return time.fromNow();
+			}
 
-		const { status, modified, date } = post;
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+			let sameElseTranslation;
+			if ( noPrefix ) {
+				sameElseTranslation = this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
+				} );
+			} else {
+				sameElseTranslation = this.props.translate( '[on] ll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" and "on" is translated',
+				} );
+			}
+
+			scheduledTime = scheduledDate.calendar( null, {
+				sameElse: sameElseTranslation,
+			} );
 		}
 
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		return scheduledDate.diff( now, 'years' ) > 0
+			? scheduledTime
+			: scheduledTime.replace( scheduledDate.format( 'Y' ), '' );
 	}
 
 	getTimeText() {
 		const time = this.getTimestamp();
-		if ( ! time ) {
-			return null;
-		}
-
 		return (
 			<span className="post-relative-time-status__time">
-				<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
-				<time className="post-relative-time-status__time-text" dateTime={ time }>
-					{ this.getDisplayedTimeFromPost( this.props.post ) }
-				</time>
+				{ time && (
+					<>
+						<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
+						<time className="post-relative-time-status__time-text" dateTime={ time }>
+							{ this.getDisplayedTimeForLabel( true ) }
+						</time>
+					</>
+				) }
 			</span>
 		);
 	}
 
-	getStatusText() {
+	/**
+	 * Get status label
+	 *
+	 * @param {boolean} onlySticky sends back the "sticky" label. Special case as is using the same template but for is unrelated to the status
```

Comment:
To avoid confusion, can we maybe separate concerns and create new functions for the sticky and pending review labels?

```
getStickyLabel() {
	return (
		<span className="post-relative-time-status__status is-sticky">
			<Gridicon icon="bookmark-outline" size={ 18 } />
			<span className="post-relative-time-status__status-text">{ this.props.translate( 'sticky' ) }</span>
		</span>
	);
}

getPendingReviewLabel() {
	return (
		<span className="post-relative-time-status__status is-pending">
			<Gridicon icon="aside" size={ 18 } />
			<span className="post-relative-time-status__status-text">{ this.props.translate( 'pending review' ) }</span>
		</span>
	);
}
```

That way we can get rid of the `onlySticky`/`onlyPending` arguments here, and used the specific functions for the labels [when needed](https://github.com/Automattic/wp-calypso/blob/00c312676cd04b2a33c26e80984ed03db27fe531/client/my-sites/post-relative-time-status/index.jsx#L199-L200): 

```
{ showPublishedStatus ? this.getStatusText() : timeText }
{ post.sticky && this.getStickyLabel() }
{ post.status === 'pending' && this.getPendingReviewLabel() }
```

Reply:
I didn't want to duplicate the HTML for maintenance purposes, but I think I found a good compromise:

https://github.com/Automattic/wp-calypso/pull/41301/files#diff-d7a7d45a786a1fd8012a5e9abc0a1d78R176-R200